### PR TITLE
feat(scalars): add TextareaField with all props and base for styles, stories, tests and behavior

### PR DESCRIPTION
**Pull request by nesdc88**
## Ticket
https://trello.com/c/uUwA1jHm/715-1-string-multi-line-component

## Description
Add TextareaField with all props and base for styles, stories, tests and behavior.

## What solved
- [X] Add TextareaField with all props.
- [X] Add base styles, stories and tests.
- [X] Add autoExpand behavior.
- [X] Add support for trim, uppercase and lowercase the text.
- [X] Code improvements.

### CODE REVIEW

File: packages/design-system/src/scalars/components/fragments/textarea-field/textarea-field.tsx [hunk 1]
```diff
@@ -1,17 +1,228 @@
...
+    },
+    ref,
+  ) => {
+    const autoCompleteValue = autoComplete ? "on" : "off";
+    const spellCheckValue = spellCheck ? "true" : "false";
+    const hasError = errors.length > 0;
+
+    const prefix = useId();
+    const id = propId ?? `${prefix}-textarea`;
+    const name = propName ?? `${prefix}-textarea`;
```
Comment (YasielCabrera) [inline packages/design-system/src/scalars/components/fragments/textarea-field/textarea-field.tsx]: the name is a required in comming updates, you can use it without prefixes or autogenerated as this should be handled by the user as it is critical for validation/form submission

File: packages/design-system/src/scalars/components/fragments/textarea-field/textarea-field.tsx [hunk 2]
```diff
@@ -1,17 +1,228 @@
...
+      trim = false,
+      uppercase = false,
+      value,
+      warnings = [],
+      ...props
+    },
+    ref,
+  ) => {
+    const autoCompleteValue = autoComplete ? "on" : "off";
+    const spellCheckValue = spellCheck ? "true" : "false";
```
Comment (YasielCabrera) [inline packages/design-system/src/scalars/components/fragments/textarea-field/textarea-field.tsx]: booleans are automatically converted to string, no need of doing this

File: packages/design-system/src/scalars/components/fragments/textarea-field/textarea-field.tsx [hunk 3]
```diff
@@ -1,17 +1,228 @@
...
+      if (autoExpand) {
+        // Reset height to allow shrinking
+        element.style.height = "auto";
+        // Set to scrollHeight to expand based on content
+        element.style.height = `${element.scrollHeight}px`;
+      }
+    };
+
+    useEffect(() => {
+      if (ref && "current" in ref && ref.current && autoExpand) {
```
Comment (YasielCabrera) [inline packages/design-system/src/scalars/components/fragments/textarea-field/textarea-field.tsx]: suggestion: `ref?.current && autoExpand` should be more readable

File: packages/design-system/src/scalars/components/fragments/textarea-field/textarea-field.test.tsx [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+      );
+
+      const textarea = screen.getByRole("textbox");
+      const counter = screen.getByText("4/10");
+
+      expect(textarea).toHaveAttribute("aria-label", "Text area");
+      expect(textarea).toHaveAttribute("aria-invalid", "true");
+      expect(counter).toBeInTheDocument();
+    });
+  }); */
```
Comment (YasielCabrera) [inline packages/design-system/src/scalars/components/fragments/textarea-field/textarea-field.test.tsx]: Please, remove this code, you can access it through the git history

**Review by YasielCabrera — CHANGES_REQUESTED** [review]

File: packages/design-system/src/scalars/components/fragments/textarea-field/textarea-field.tsx (hunk 1 above)
Comment (nesdc88) [reply packages/design-system/src/scalars/components/fragments/textarea-field/textarea-field.tsx]: Done

File: packages/design-system/src/scalars/components/fragments/textarea-field/textarea-field.tsx (hunk 2 above)
Comment (nesdc88) [reply packages/design-system/src/scalars/components/fragments/textarea-field/textarea-field.tsx]: Done

File: packages/design-system/src/scalars/components/fragments/textarea-field/textarea-field.tsx (hunk 3 above)
Comment (nesdc88) [reply packages/design-system/src/scalars/components/fragments/textarea-field/textarea-field.tsx]: I applied a different and more readable approach

File: packages/design-system/src/scalars/components/fragments/textarea-field/textarea-field.test.tsx (hunk 4 above)
Comment (nesdc88) [reply packages/design-system/src/scalars/components/fragments/textarea-field/textarea-field.test.tsx]: Done

**Review by YasielCabrera — APPROVED** [review]
lgtm

**Pull request merged**